modules/substrate/src/verifier.rs

```diff
@@ -127,19 +130,27 @@ where
 			return Err(ImportError::InvalidChildNumber);
 		}
 
+		let scheduled_change = find_scheduled_change(&header);
+
 		// This header requires a justification since it enacts an authority set change. We don't
 		// need to act on it right away (we'll update the set once this header gets finalized), but
 		// we need to make a note of it.
 		//
-		// TODO: This assumes that we can only have one authority set change pending at a time.
-		// This is not strictly true as Grandpa may schedule multiple changes on a given chain
-		// if the "next next" change is scheduled after the "delay" period of the "next" change
+		// Note: This assumes that we can only have one authority set change pending at a time.
+		// While this is not strictly true of Grandpa (it can have multiple pending changes, even
+		// across forks), this assumption simplifies our tracking of authority set changes.
 		let requires_justification = if let Some(change) = self.storage.scheduled_set_change() {
+			// We don't want to accept a header which schedules an authority set change
```

Comment:
It's still a draft - just wanted to clarify. I thought that we're going to support single pending change **per fork**, no? In current implementation it seems that anyone could craft an invalid (because we don't verify headers) header that is scheduling some change and then pallet will stuck waiting for justification of this/next header forever. I thought that we're not going to have more than 1 scheduled change per fork - i.e. if this fork needs justification for header `H`, then we'll reject all headers that are trying to continue this fork until we'll have a justification for `H`. At the same time, the other fork may still be active.

Reply:
Yeah there's still some confusion around this on my end. You're right, we should only allow one pending change _per fork_.